clang/lib/Sema/SemaChecking.cpp

```diff
@@ -5935,8 +5936,14 @@ static void CheckFormatString(
     llvm::SmallBitVector &CheckedVarArgs, UncoveredArgHandler &UncoveredArg,
     bool IgnoreStringsWithoutSpecifiers);
 
-static const Expr *maybeConstEvalStringLiteral(ASTContext &Context,
-                                               const Expr *E);
+enum StringLiteralConstEvalResult {
+  SLCER_NotEvaluated,
+  SLCER_NotNullTerminated,
+  SLCER_Evaluated,
+};
+
+static StringLiteralConstEvalResult
+constEvalStringAsLiteral(Sema &S, const Expr *E, const StringLiteral *&SL);
```

Comment:
```suggestion
enum StringLiteralConstEvalResult {
  SLCER_NotEvaluated,
  SLCER_NotNullTerminated,
  SLCER_Evaluated,
};

static StringLiteralConstEvalResult
EvaluateStringAndCreateLiteral(Sema &S, const Expr *E, const StringLiteral *&SL);
```

Reply:
We should really reflect we might create a scratch space here.
Also, can we comment this function?